File: 1-two-sum/two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numToIndex = [Int: Int]()
        
        for (index, num) in nums.enumerated() {
            let complement = target - num
            
            if let complementIndex = numToIndex[complement] {
                return [complementIndex, index]
            }
            
            numToIndex[num] = index
        }
        
        // Since problem guarantees exactly one solution, we'll always return above
        return []
    }
}

File: 1-two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val numToIndex = mutableMapOf<Int, Int>()
        
        for ((index, num) in nums.withIndex()) {
            val complement = target - num
            
            if (numToIndex.containsKey(complement)) {
                return intArrayOf(numToIndex[complement]!!, index)
            }
            
            numToIndex[num] = index
        }
        
        // Since problem guarantees exactly one solution, we'll always return above
        return intArrayOf()
    }
}
